api/tests/test_audit_log.py

```diff
@@ -0,0 +1,176 @@
+from django.test import Client
+from django.urls import reverse
+
+from api.factories import DeliveryLogFactory
+from api.models import DeliveryLog
+from audit_log.enums import Operation
+from audit_log.models import AuditLogEntry
+
+
+def test_visiting_delivery_log_admin_view_writes_read_log(admin_user):
+    """
+    Test that visiting the DeliveryLog admin list view writes a read log for each object
+    """
+    client = Client()
+    client.force_login(admin_user)
+
+    delivery_logs = DeliveryLogFactory.create_batch(3)
+
+    # Visit the admin list page of DeliveryLog (/admin/api/deliverylog/)
+    url = reverse("admin:api_deliverylog_changelist")
+    response = client.get(url)
+    assert response.status_code == 200
+
+    # Assert that an AuditLogEntry was created for each DeliveryLog object
+    qs = AuditLogEntry.objects.filter(message__audit_event__target__path=url)
+    # There should be just 1 audit log entry
+    assert qs.count() == 1
+    audit_log_entry = qs.first()
+    # The audit log entry should have all the delivery log ids in it
+    assert audit_log_entry.message["audit_event"]["target"]["object_ids"] == [
+        str(log.pk) for log in delivery_logs
+    ]
```

Comment:
If you want to add a bit more resiliency to this test you could test that the sorted lists match i.e. the order of the IDs wouldn't be tested for, only that the IDs themselves are the same i.e.:
```suggestion
    assert sorted(
        audit_log_entry.message["audit_event"]["target"]["object_ids"]
    ) == sorted(str(log.pk) for log in delivery_logs)
```
unless you see the order having significance.

Reply:
changed. It was actually needed after the fixes and using the pagination object